File: javascript/1051.Height-Checker.js
/**
 * @param {number[]} heights
 * @return {number}
 */
var heightChecker = function (heights) {
  const count = new Array(101).fill(0);
  let current_height = 1;
  let mismatches = 0;

  for (const height of heights) {
    count[height] += 1;
  }

  for (let i = 0; i < heights.length; i++) {
    while (count[current_height] === 0) {
      current_height += 1;
    }

    if (heights[i] !== current_height) {
      mismatches += 1;
    }

    count[current_height] -= 1;
  }

  return mismatches;
};

File: python/1051.Height-Checker.py
class Solution:
    def heightChecker(self, heights: List[int]) -> int:
        count = [0] * 101
        mismatches = 0
        current_height = 1

        for height in heights:
            count[height] += 1

        for i in range(len(heights)):
            while count[current_height] == 0:
                current_height += 1

            if heights[i] != current_height:
                mismatches += 1

            count[current_height] -= 1

        return mismatches